### PR TITLE
Fixed sync target detection

### PR DESCRIPTION
**Pull request by arkpar**
This fixes sync target detection. With the current mean selection method it is possible for this function to return a block that's lower than our best block. This PR changes the mean selection to only consider peers that are ahead. 

See #11547

Fixes https://github.com/paritytech/devops/issues/1767

### CODE REVIEW

**Review by cheme — APPROVED** [review]
Oh I thought #11705 was merged. So this PR does the same with best_queued_number instead of actual best. Sounds correct.

**Comment by nazar-pc** [comment]
But queued blocks aren't guaranteed to import successfully, I'm wondering if #11705 that checks just the block successfully imported is a better alternative.

And yeah, #11705 was open for over 3 weeks now :disappointed:

**Comment by arkpar** [comment]
I also forgot https://github.com/paritytech/substrate/pull/11705 exists. From sync POV all blocks added to the import queue are considered to be "synced". If there's a error with a block import at later stage, the sync process is reset. So I think using `best_queued_number` is better here for consistency.

File: client/network/sync/src/lib.rs [hunk 1]
```diff
@@ -1721,7 +1721,17 @@ where
 
 	/// Returns the best seen block.
 	fn best_seen(&self) -> Option<NumberFor<B>> {
-		let mut best_seens = self.peers.values().map(|p| p.best_number).collect::<Vec<_>>();
+		let mut best_seens = self
+			.peers
+			.values()
+			.filter_map(|p| {
+				if p.best_number > self.best_queued_number {
```
[the remark this answers is not in the document]
Comment (arkpar) [reply client/network/sync/src/lib.rs]: Or rather return `None` if the median is in the past.

**Comment by arkpar** [comment]
bot merge

**Comment by nazar-pc** [comment]
`master` needs to be merged in after https://github.com/paritytech/substrate/pull/11829

**Comment by arkpar** [comment]
bot merge

**Pull request merged**